core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -5204,7 +5204,11 @@ public static List compact(List list) {
   public static List arrayAppend(List list, Object element) {
     final List result = new ArrayList(list.size() + 1);
     result.addAll(list);
-    result.add(element);
+    if (element instanceof Byte) {
+      result.add(((Byte) element).intValue());
```

Comment:
This doesn't seem right. Is the array always containing integers?
What happens for an array of BIGINT?
What happens for Short?

Reply:
Thanks I will try other methods

Reply:
+1 can't do this. we must make sure when you call `arrayAppend` the input `List` already is a isomorphic result list (a result list has same operand's type). so it means we need an explicit casting before you call SqlFunctions.